### PR TITLE
build: Django3.2 upgrade

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#56 

**Reviewer Note:** 🔴 
In this PR I've used some other branch edx-platform than master to run the tests on as you can see [here](https://github.com/mitodl/edx-sysadmin/pull/57/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR31). The reason I'm doing that is that we have been using a test configuration `test_static_optimized`(Ref: https://github.com/mitodl/edx-sysadmin/blob/main/run_devstack_integration_tests.sh#L15) to generate our assets to test some of our functionality which now breaks because of import errors that are still not fixed Open edX's master. Although this is just the test config and doesn't really affect the actual functionality of the plugin.
I only did the above so it's easy for the reviewer to see the details of the passing tests. I'll set the platform's branch back to master upon merging this after the review/approval because in either case, the fix will be going to the master branch of the platform.

#### What's this PR do?
- Upgrades the package to use Django version 3.2 which is the same as edX.
- In the process of doing the above I've also run codemods edX's from [Open edX's repo-tools](https://github.com/edx/repo-tools/tree/master/edx_repo_tools/codemods/django3)
- Updated the tox files, requirements as of Django3.2
- Fixes/Ignores the warnings

#### How should this be manually tested?
- First off check that the tests are passing
- Following the instructions in the repo readme, install the sysadmin plugin
- Verify that you are able to access the sysadmin and able to test the functionality e.g. `Users Creation, Course Deletion, Course Imports and Git Logs`.

#### Where should the reviewer start?
Plugin installation

#### Screenshots (if appropriate)
Screenshot 2021-11-23 at 5 28 15 PM
Screenshot 2021-11-23 at 5 28 44 PM
Screenshot 2021-11-23 at 5 28 58 PM
Screenshot 2021-11-23 at 5 29 11 PM

### CODE REVIEW

**Comment by arslanashraf7** [comment]
FYI, For **anyone** reviewing this PR that I've added a detailed reviewer note in the PR description which you might want to take a look at. I'm cross-posting it here as well:
>Reviewer Note: 🔴
In this PR I've used some other branch edx-platform than master to run the tests on as you can see here. The reason I'm doing that is that we have been using a test configuration test_static_optimized(Ref: https://github.com/mitodl/edx-sysadmin/blob/main/run_devstack_integration_tests.sh#L15) to generate our assets to test some of our functionality which now breaks because of import errors that are still not fixed Open edX's master. Although this is just the test config and doesn't really affect the actual functionality of the plugin.
I only did the above so it's easy for the reviewer to see the details of the passing tests. I'll set the platform's branch back to master upon merging this after the review/approval because in either case, the fix will be going to the master branch of the platform.

File: tox.ini [hunk 1]
```diff
@@ -1,39 +1,21 @@
...
-; D405 = Section name should be properly capitalized (numpy style)
-; D406 = Section name should end with a newline (numpy style)
-; D407 = Missing dashed underline after section (numpy style)
-; D408 = Section underline should be in the line following the section’s name (numpy style)
-; D409 = Section underline should match the length of its name (numpy style)
-; D410 = Missing blank line after section (numpy style)
-; D411 = Missing blank line before section (numpy style)
-; D412 = No blank lines allowed between a section header and its content (numpy style)
-; D413 = Missing blank line after last section (numpy style)
-; D414 = Section has no content (numpy style)
```
Comment (umarmughal824) [inline tox.ini:28]: why these lines are removed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ jobs:
           cd ../edx-platform
           git remote add mitodl https://github.com/mitodl/edx-platform.git
           git fetch mitodl
-          git checkout mitodl/mitx/lilac
+          git checkout mitodl/arslan/fix-pipeline-import
```
Comment (umarmughal824) [inline .github/workflows/ci.yml:31]: do we have maple branch ready to use here?

File: tox.ini (hunk 1 above)
Comment (arslanashraf7) [reply tox.ini:28]: These were removed through one of the Codemod scripts(https://github.com/edx/repo-tools/blob/master/edx_repo_tools/codemods/django3/tox_modernizer.py) that we run as part of edX's standards. If we need to ignore these warnings moving on we can add them back.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (arslanashraf7) [reply .github/workflows/ci.yml:31]: We do have a Maple release but that would not put any impact on fixing the tests as I mentioned in the reviewer notes about why. https://github.com/mitodl/edx-sysadmin/pull/57#issuecomment-976476851

**Review by umarmughal824 — APPROVED** [review]
looks good 👍

**Pull request merged**